cli/pkg/kctrl/cmd/package/init/init.go

```diff
@@ -254,22 +257,50 @@ func (o *InitOptions) updatePackageInstall(pkgInstall *pkgv1alpha1.PackageInstal
 	}
 }
 
-func (o *InitOptions) updatePackage(pkg *v1alpha1.Package, pkgBuild *PackageBuild) {
+func (o *InitOptions) updatePackage(pkg *v1alpha1.Package, pkgBuild *PackageBuild) error {
 	if len(pkg.Spec.Version) == 0 {
 		pkg.Spec.Version = "0.0.0"
 	}
 	pkg.Name = pkg.Spec.RefName + "." + pkg.Spec.Version
 
 	if pkg.Spec.Template.Spec == nil {
 		pkg.Spec.Template.Spec = &kcv1alpha1.AppSpec{}
-	}
-
-	pkg.Spec.Template.Spec.Fetch = []kcv1alpha1.AppFetch{{Git: &kcv1alpha1.AppFetchGit{}}}
-	pkg.Spec.Template.Spec.Template = pkgBuild.GetAppSpec().Template
-
-	if len(pkg.Spec.Template.Spec.Deploy) == 0 {
+		pkg.Spec.Template.Spec.Fetch = []kcv1alpha1.AppFetch{{Git: &kcv1alpha1.AppFetchGit{}}}
+		pkg.Spec.Template.Spec.Template = pkgBuild.GetAppSpec().Template
 		pkg.Spec.Template.Spec.Deploy = pkgBuild.GetAppSpec().Deploy
+	} else {
+		if !isAppSpecSame(pkg, pkgBuild) {
+			o.ui.PrintInformationalText("AppSpec section of Package(inside package-resources.yml) and " +
+				"PackageBuild(inside package-build.yml) is different. " +
+				"Either choose to overwrite the Package AppSpec or leave as it is.")
+			overrideOptions := []string{"Yes", "No"}
+			choiceOpts := ui.ChoiceOpts{
+				Label:   "Overwrite the Package AppSpec from PackageBuild",
+				Default: 1,
+				Choices: overrideOptions,
+			}
+			selectedIndex, err := o.ui.AskForChoice(choiceOpts)
```

Comment:
Should this not be ask for confirmation rather than `AskForChoice`, since the options are yes/no

Reply:
`AskForChoice` if you select `no`, it exits today.

Reply:
I think it returns an error today I believe, and that error if returned causes the exit.
Instead we could respond differently if there is an error.

Just suggesting this because putting "y" or "n" might be intuitive for a few folks

Reply:
Thing is non interactive flow, `AskForConfirmation` will always return true and we want it to be false. Also, with `AskForChoice`, user will be entering `0` or `1` rather than `Yes` or `No`. If you still think we should move to `AskForConfirmation`, we can rephrase the question and do the error handling for `false` use case.